### PR TITLE
clarify TFA is the best authentication method

### LINKED ISSUE #372: Update recommendations on authentication methods
Story:
As a developer I want updated recommendations on authentication methods so that I can use best practices.

[David's comment on a recent PR](https://github.com/pantheon-systems/documentation/pull/302#issuecomment-85672363) states that LDAP is no longer best practice.

We need to:

 - [ ] Identify which docs require an update
 - [ ] Determine the best practice recommendations for authentication methods and SSO

### PR DESCRIPTION
**Pull request by nataliejeremy**
Closes #372 

@bmackinney @rachelwhitton Please review.

### CODE REVIEW

File: source/docs/articles/sites/code/sso-and-identity-federation.md [hunk 1]
```diff
@@ -11,6 +11,11 @@ Pantheon’s flexible infrastructure does not restrict protocols or ports used f
 
 Use SSL certificates for encrypted secure communication with externally hosted servers for authentication.
 
+## Two-Factor Authentication
+Two-factor authentication (TFA) is a security practice that requires users of your website to provide, along with their standard username and password, an additional form of authentication to log in.
+
+We strongly recommend using SAML (both for sites and the Dashboard) and OneLogin. Learn more about [two-factor authentication](/docs/guides/two-factor-authentication/).
```
Comment (davidstrauss) [inline source/docs/articles/sites/code/sso-and-identity-federation.md]: We should strongly recommend using SAML but not OneLogin. There are multiple legitimate options for SAML IdP out there.

File: source/docs/articles/sites/code/ldap-and-ldaps.md [hunk 2]
```diff
@@ -10,9 +10,9 @@ keywords: ldap, ldaps, ldap and ldaps, using ldap as a provider, ldap provider,
...
 
-LDAP provides access and maintenance of a distributed directory storing organized sets of records. LDAP is often used for single sign-on authentication, and many sites on Pantheon use LDAP and LDAPS for authentication. Authentication requests will originate from Pantheon to the LDAP server used (outbound from Pantheon).  
+LDAP provides access and maintenance of a distributed directory storing organized sets of records. LDAP is often used for single sign-on authentication and authentication requests will originate from Pantheon to the LDAP server used (outbound from Pantheon). However, we do not recommend using LDAP for single sign-on authentication as it is not secure.  The recommended approach is SAML (both for sites and the Dashboard) and OneLogin. Learn more about [two-factor authentication](/docs/guides/two-factor-authentication/).
```
Comment (rachelwhitton) [inline source/docs/articles/sites/code/ldap-and-ldaps.md]: @nataliejeremy what do you think of reducing this paragraph a bit and wrapping it in the alert styling? Something like this: 
screen shot 2015-09-08 at 3 29 24 pm

Comment (nataliejeremy) [inline source/docs/articles/sites/code/ldap-and-ldaps.md]: That gives the impression it's not OK to use, when it is. We just don't recommend it. @bmackinney what do you think?

Comment (rachelwhitton) [inline source/docs/articles/sites/code/ldap-and-ldaps.md]: @nataliejeremy I may be misunderstanding the [original comments](https://github.com/pantheon-systems/documentation/pull/302#issuecomment-85672363), but I don't think LDAP for SSO can be used at all unless it's in conjunction with a SaaS provider (such as onelogin). If so, we should update the paragraph to suggest they use SAML with a SaaS provider to bridge authentication over to LDAP. (Current verbiage doesn't mention bridging back to LDAP and mentions OneLogin directly). 

If a user is on this doc hoping to configure some type of SSO system with LDAP, I think a warning alert is fine. The verbiage used within the warning should be enough to inform the reader it doesn't apply to all implementations of LDAP, but just SSO. 

I defer to you and @bmackinney on that though

Comment (nataliejeremy) [inline source/docs/articles/sites/code/ldap-and-ldaps.md]: They would need an IdP provider to use LDAP. But that content already existed and I don't think we want to remove it or give them further instructions on using LDAP for SSO. IMO it doesn't warrant the warning treatment.

If I understood the issue correctly, we just want to make it clear we recommend SAML. Maybe @ari-gold or @bmackinney can clarify this.

Comment (nataliejeremy) [inline source/docs/articles/sites/code/ldap-and-ldaps.md]: @rachelwhitton Just updated to show the info as a note, per convo with @bmackinney

**Comment by rachelwhitton** [comment]
Awesome, thanks @nataliejeremy! 

+1

**Pull request merged**